src/SimpleGraphs/generators/staticgraphs.jl

```diff
@@ -536,6 +536,60 @@ function double_binary_tree(k::Integer)
     return g
 end
 
+"""
+    regular_tree([T=Int64, ], k::Integer, z::integer)
+
+Create a regular tree or [perfect z-ary tree](https://en.wikipedia.org/wiki/M-ary_tree#Types_of_m-ary_trees): 
+a `k`-level tree where all nodes except the leaves have exactly `z` children. 
+For `z = 2` one recovers a binary tree.
+
+# Examples
+```jldoctest
+julia> regular_tree(4, 3)
+{40, 39} undirected simple Int64 graph
+
+julia> regular_tree(Int8, 3, 2)
+{7, 6} undirected simple Int8 graph
+
+julia> regular_tree(5, 2) == binary_tree(5)
+true
+```
+"""
+function regular_tree(T::Type{<:Integer}, k::Integer, z::Integer)
+    z <= 0 && throw(DomainError(z, "number of children must be positive"))
+    z == 1 && return path_graph(T(k))
+    k <= 0 && return SimpleGraph(zero(T))
+    k == 1 && return SimpleGraph(one(T))
+    if Graphs.isbounded(k) && (BigInt(z)^k - 1) ÷ (z - 1) > typemax(T)
+        throw(InexactError(:convert, T, (BigInt(z)^k - 1) ÷ (z - 1)))
+    end
+
+    n = T((z^k - 1) / (z - 1))
+    ne = n - 1
+    fadjlist = Vector{Vector{T}}(undef, n)
+    @inbounds fadjlist[1] = convert.(T, 2:(z + 1))
+    @inbounds for l in 2:(k - 1)
+        w = (z^(l - 1) - 1) ÷ (z - 1)
+        x = w + z^(l - 1)
+        @simd for i in 1:(z^(l - 1))
+            j = w + i
+            fadjlist[j] = [
+                T(ceil((j - x) / z) + w)
+                convert.(T, (x + (i - 1) * z + 1):(x + i * z))
+            ]
+        end
+    end
+    l = k
+    w = (z^(l - 1) - 1) ÷ (z - 1)
+    x = w + z^(l - 1)
+    @inbounds @simd for j in (w + 1):x
+        fadjlist[j] = T[ceil((j - x) / z) + w]
+    end
+    return SimpleGraph(ne, fadjlist)
+end
+
```

Comment:
I would add the docstring also to this function. 
````suggestion

"""
    regular_tree(k, z)

Create a regular tree or [perfect z-ary tree](https://en.wikipedia.org/wiki/M-ary_tree#Types_of_m-ary_trees): 
a `k`-level tree where all nodes except the leaves have exactly `z` children. 
For `z = 2` one recovers a binary tree.
# Examples
```jldoctest
julia> regular_tree(4, 3)
{40, 39} undirected simple Int64 graph
julia> regular_tree(5, 2) == binary_tree(5)
true
```
"""
````

Reply:
It's quite often the case, that the docstrings for multiple methods of a function are only added for a single method - I think that would also be better here, than adding docstrings for both methods of this function.